modules/core/src/main/java/org/apache/ignite/internal/processors/cache/binary/CacheObjectBinaryProcessorImpl.java

```diff
@@ -535,8 +535,14 @@ public GridBinaryMarshaller marshaller() {
 
             BinaryMetadata mergedMeta = BinaryUtils.mergeMetadata(oldMeta, newMeta0, changedSchemas);
 
-            if (oldMeta != null && mergedMeta == oldMeta && metaHolder.pendingVersion() == metaHolder.acceptedVersion())
-                return; // Safe to use existing schemas.
+            if (mergedMeta == oldMeta)
+                // Metadata locally is up-to-date. Waiting for updating metadata in an entire cluster, if necessary.
+                while (true) {
```

Comment:
Instead of spinning in loop we should use BinaryMetadataTransport::awaitMetadataUpdate method and wait when metaHolder.pendingVersion is accepted by the cluster.